src/utility/Albany_KokkosUtils.hpp

```diff
@@ -1,53 +1,103 @@
-#ifndef ALBANY_KOKKOS_UTILS_HPP
-#define ALBANY_KOKKOS_UTILS_HPP
-
-// Get KOKKOS_INLINE_FUNCTION
-#include "Kokkos_Macros.hpp"
-
-// Get Sacado max/min
-#include "Sacado.hpp"
-
-namespace KU
-{
-
-#ifdef KOKKOS_ENABLE_CUDA
-
-template <typename T>
-KOKKOS_INLINE_FUNCTION
-const T& max (const T& a, const T& b) { return a > b ? a : b; }
-
-template <typename T>
-KOKKOS_INLINE_FUNCTION
-const T& min (const T& a, const T& b) { return a < b ? a : b; }
-
-inline bool
-IsNearDeviceMemoryLimit()
-{
-  size_t max_free_t = 104857600; // "Near" is subjective, let's use 100 MiB
-  size_t free_t, total_t;
-  cudaMemGetInfo(&free_t,&total_t);
-  return free_t < max_free_t;
-}
-
-#else
-
-using std::max;
-using std::min;
-
-inline bool
-IsNearDeviceMemoryLimit()
-{
-  return false;
-}
-
-#endif
-
-using Sacado::Fad::max;
-using Sacado::Fad::min;
-using Sacado::Fad::Exp::max;
-using Sacado::Fad::Exp::min;
-
-}
-
-#endif // ALBANY_KOKKOS_UTILS_HPP
-
+#ifndef ALBANY_KOKKOS_UTILS_HPP
+#define ALBANY_KOKKOS_UTILS_HPP
+
+// Get KOKKOS_INLINE_FUNCTION
+#include "Kokkos_Macros.hpp"
+
+// Get Sacado max/min
+#include "Sacado.hpp"
+
+namespace KU
+{
+
+// Check device memory
+#ifdef KOKKOS_ENABLE_CUDA
+inline bool
+IsNearDeviceMemoryLimit()
+{
+  size_t max_free_t = 104857600; // "Near" is subjective, let's use 100 MiB
+  size_t free_t, total_t;
+  cudaMemGetInfo(&free_t,&total_t);
+  return free_t < max_free_t;
+}
+#else
+inline bool
+IsNearDeviceMemoryLimit()
+{
+  return false;
+}
+#endif
+
+// Kernel min/max
+#ifdef KOKKOS_ENABLE_CUDA
+template <typename T>
+KOKKOS_INLINE_FUNCTION
+const T& max (const T& a, const T& b) { return a > b ? a : b; }
+
+template <typename T>
+KOKKOS_INLINE_FUNCTION
+const T& min (const T& a, const T& b) { return a < b ? a : b; }
+#else
+using std::max;
+using std::min;
+#endif
+using Sacado::Fad::max;
+using Sacado::Fad::min;
+using Sacado::Fad::Exp::max;
+using Sacado::Fad::Exp::min;
+
+// Choose atomic based on execution space
+template <typename ExeSpace>
+struct IsAtomic
```

Comment:
Very subjective (i.e., feel free to leave like this): I think `NeedsAtomic` would be more suited than `IsAtomic`, since it makes no sense to say that the exe space is atomic.

Reply:
This makes sense, thanks! I'll make the change in push. I tested this on clang/weaver and it works fine.